packages/lexical/src/LexicalNode.ts

```diff
@@ -198,6 +232,19 @@ export class LexicalNode {
   __prev: null | NodeKey;
   /** @internal */
   __next: null | NodeKey;
+  /** @internal */
+  __state: DeepImmutable<State> = {};
```

Comment:
A more precise type would be something like this. `__state = 12` should not be valid.

```suggestion
  __state: { [k in string]?: DeepImmutable<State> } = {};
```

Reply:
You are confusing the `State` and `StateValue` types. As it stands right now `__state = 12` is invalid